packages/@react-aria/collections/src/Document.ts

```diff
@@ -177,7 +181,11 @@ export class BaseNode<T> {
     if (child.parentNode !== this || !this.ownerDocument.isMounted) {
       return;
     }
-    
+
+    if (this._minInvalidChildIndex === child && child.previousSibling) {
+      this.invalidateChildIndices(child.previousSibling);
```

Comment:
Shouldn't it move to the next sibling? After removing a child, the next sibling will now have a new index. The previous sibling won't change. Also what happens if you remove the first child?

Reply:
actually I see I already had that below. I think we can just set minInvalidChildIndex to null here.